Framework/PythonInterface/mantid/api/src/Exports/Algorithm.cpp

```diff
@@ -76,6 +89,87 @@ void translateCancel(const Algorithm::CancelException &exc) {
   UNUSED_ARG(exc);
   PyErr_SetString(PyExc_KeyboardInterrupt, "");
 }
+
+template <typename T> boost::optional<T> extractArg(ssize_t index, const tuple &args) {
+  if (index < len(args)) {
+    return boost::optional<T>(extract<T>(args[index]));
+  }
+  return boost::none;
+}
+
+template <typename T> void extractKwargs(const dict &kwargs, const std::string &keyName, boost::optional<T> &out) {
+  if (kwargs.has_key(keyName)) {
+    out = boost::optional<T>(extract<T>(kwargs.get(keyName)));
+  }
+}
+
+class SetPropertyVisitor final : public Mantid::PythonInterface::IPyTypeVisitor {
+public:
+  SetPropertyVisitor(Mantid::API::Algorithm_sptr &alg, std::string const &propName)
+      : m_alg(alg), m_propName(propName) {}
+
+  void operator()(bool value) const override { setProp(value); }
+  void operator()(long value) const override { setProp(static_cast<int>(value)); }
+  void operator()(double value) const override { setProp(value); }
+  void operator()(std::string value) const override { m_alg->setPropertyValue(m_propName, value); }
+
+  void operator()(std::vector<bool> value) const override { setProp(value); }
+  void operator()(std::vector<long> value) const override { setProp(value); }
+  void operator()(std::vector<double> value) const override { setProp(value); }
+  void operator()(std::vector<std::string> value) const override { setProp(value); }
+
+  using Mantid::PythonInterface::IPyTypeVisitor::operator();
+
+private:
+  template <typename T> void setProp(const T &val) const { m_alg->setProperty(m_propName, val); }
+
+  Mantid::API::Algorithm_sptr &m_alg;
+  std::string const &m_propName;
+};
+
+// Signature createChildWithProps(self, name, startProgress, endProgress, enableLogging, version, **kwargs)
+object createChildWithProps(tuple args, dict kwargs) {
+  Mantid::API::Algorithm_sptr parentAlg = extract<Mantid::API::Algorithm_sptr>(args[0]);
+  auto name = extractArg<std::string>(1, args);
+  auto startProgress = extractArg<double>(2, args);
+  auto endProgress = extractArg<double>(3, args);
+  auto enableLogging = extractArg<bool>(4, args);
+  auto version = extractArg<int>(5, args);
+
+  const std::array<std::string, 5> reservedNames = {"name", "startProgress", "endProgress", "enableLogging", "version"};
+
+  extractKwargs<std::string>(kwargs, reservedNames[0], name);
```

Comment:
Does this let me overwrite the args with kwargs? I could then change the algorithm name to something else right, i.e.

```
dataX = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0]
dataY = [1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0,11.0,12.0]
args={"DataX": dataX, "DataY": dataY, "NSpec":4, 'name': "FitPeaks"}
parent_alg = AlgorithmManager.createUnmanaged('Load')
child_alg = parent_alg.createChildAlgorithm('CreateWorkspace', 0.0, version=1,**args)
child_alg.execute()
```
would begin to create the `CreateWorkspace` algorithm, but it would end up creating the fit algorithm due to the kwargs. Maybe we should only overwrite things if the optionals are empty

Reply:
Good spot! I think we should throw an error if the user attempts to set the same property twice. This works if you try to set a kwarg twice - I get the error `got multiple values for keyword argument 'name'` - we've just not taken it into account with a mix of positional and kwargs setting the same thing.

Reply:
Hmm, I'm not sure the best way to solve this :thinking: 
Because we've made name effectively mandatory as a positional arg, by hardcoding arg 1 to a string.

Maybe we simply disable the name kwarg and emit a warning for that?